src/lib.rs

```diff
@@ -454,31 +454,31 @@ impl From<io::Error> for GooseError {
 #[derive(Clone)]
 pub struct GooseAttack {
     /// An optional task to run one time before starting users and running task sets.
-    test_start_task: Option<GooseTask>,
+    pub test_start_task: Option<GooseTask>,
     /// An optional task to run one time after users have finished running task sets.
-    test_stop_task: Option<GooseTask>,
+    pub test_stop_task: Option<GooseTask>,
     /// A vector containing one copy of each GooseTaskSet that will run during this load test.
-    task_sets: Vec<GooseTaskSet>,
+    pub task_sets: Vec<GooseTaskSet>,
     /// A checksum of the task_sets vector to be sure all workers are running the same load test.
-    task_sets_hash: u64,
+    pub task_sets_hash: u64,
     /// A weighted vector containing a GooseUser object for each user that will run during this load test.
-    weighted_users: Vec<GooseUser>,
+    pub weighted_users: Vec<GooseUser>,
     /// An optional default host to run this load test against.
-    host: Option<String>,
+    pub host: Option<String>,
     /// Configuration object managed by StructOpt.
-    configuration: GooseConfiguration,
+    pub configuration: GooseConfiguration,
     /// By default launch 1 user per number of CPUs.
-    number_of_cpus: usize,
+    pub number_of_cpus: usize,
     /// Track how long the load test should run.
-    run_time: usize,
+    pub run_time: usize,
     /// Track total number of users to run for this load test.
-    users: usize,
+    pub users: usize,
     /// Track how many users are already loaded.
-    active_users: usize,
+    pub active_users: usize,
     /// All requests statistics merged together.
-    merged_requests: GooseRequestStats,
+    pub statistics: GooseRequestStats,
     /// When the load test started.
-    started: Option<time::Instant>,
+    pub started: Option<time::Instant>,
```

Comment:
I am a fan of public properties, but in PHP world I learned that that says to users that they can rely on a stable structure of the struct.

It is like an API / ABI then.

It’s also the sole reason for getters / setters to hide the internal structure.

Just some food for thought. I am not sure how Rust handles that in general.

Reply:
This has been reworked so you have to call `get_stats()` on the `GooseAttack` internal state, which then exposes the relevant statistics. 

At this time `GooseAttack` doesn't use `GooseStats` for anything, it merely exposes the information to anyone needing it (and is currently only used by one integration test to validate that statistics are accurate).